### PR TITLE
Disable sync points in certain circumstances.

### PR DESCRIPTION
**Pull request by clumens**
This only affects sync points on IPC messages containing multiple update messages, and only when those messages are received by an attrd that does not support multiple update messages.  Hopefully, this is a pretty small number of installations.

In these cases, right now we act like sync points are supported but they are not.  For local sync points, what happens is we iterate over each individual update message.  Each update message will have a sync point attribute, and the server will send the ACK for each as it processes the message.  However, the client is only going to wait for the first ACK before continuing.  For all the other ACKs, there will not be a client listening.

Cluster-wide sync points likely work the same way, though I have not tested it to be sure.  I don't see how they could work, at least.

I don't see any way to actually handle these, so my "fix" here is to simply disable support.  Log a message and remove the attributes from the XML messages so the server doesn't try to do anything.

Fixes T601

### CODE REVIEW

**Comment by clumens** [comment]
I tried to figure out a way to solve this a while ago, but I no longer remember the problems involved and why I decided there wasn't a good fix.  So, I'm trying a new approach and just disabling it so we don't offer something we can't deliver.  Note that I have compiled this but not tested it.

File: daemons/attrd/attrd_ipc.c [hunk 1]
```diff
@@ -443,6 +457,11 @@ attrd_client_update(pcmk__request_t *request)
              */
             xmlNode *orig_xml = request->xml;
 
+            if (attrd_request_has_sync_point(request->xml)) {
+                crm_warn("Sync points in multi-update IPC messages are not supported on this cluster");
```
[the remark this answers is not in the document]
Comment (clumens) [reply daemons/attrd/attrd_ipc.c:461]: Yeah, we do need to ACK.  And in fact, it looks like if this is the approach we want to take, we need to do it earlier.  We probably need to do it in `handle_update_request` so we bypass all the waitlist stuff, too.

File: daemons/attrd/attrd_messages.c [hunk 2]
```diff
@@ -180,6 +206,14 @@ handle_update_request(pcmk__request_t *request)
         return NULL;
 
     } else {
+        /* See comment on identical block in handle_clear_failure_request */
+        if (request->xml->children != NULL && ATTRD_SUPPORTS_MULTI_MESSAGE(minimum_protocol_version) &&
+            attrd_request_has_sync_point(request->xml)) {
+            crm_warn("Ignoring sync point in request from %s because not all nodes support it",
+                     pcmk__request_origin(request));
+            remove_sync_point_attribute(request->xml, NULL);
+        }
```
Comment (kgaillot) [inline daemons/attrd/attrd_messages.c:215]: might as well functionize it (or make it part of attrd_request_has_sync_point())

**Review by kgaillot — COMMENTED** [review]
looks good to me

File: daemons/attrd/attrd_messages.c (hunk 2 above)
Comment (clumens) [reply daemons/attrd/attrd_messages.c]: Functionized.  I still need to test this out for real before I'm willing to remove the draft status.

**Comment by clumens** [comment]
It sure would be convenient for testing if we had somewhere that used multi-update messages, aside from the last-failure and fail-count stuff which seems pretty buried.

**Comment by clumens** [comment]
Tested, and the sync points get removed from the XML messages and there's no mention of sync points being hit in the logs.

**Comment by kgaillot** [comment]
retest this please

**Pull request merged**